### DISCUSSION TITLE
Generate a TOC for notebooks / pages with tags

### DISCUSSION THREAD

**Discussion by fmaussion** [executablebooks/meta#1128] [Ideas]
We use jupyter-book to host a list of tutorials for our glacier model. With time the number of tutorials has grown, and it has become increasingly difficult for new users to find their way among the large number of notebooks.

An idea that was suggested is to organize notebooks by "tags", which would allow users to browse content either with the traditional TOC (the one we use in the book), or by tags. This is a common feature in jekyll blogs, where you can browse posts [chronologically](https://oggm.org/news/) or by [tags](https://oggm.org/search/). 

I wonder if this would be difficult to implement - or perhaps it's already there? We would need a new way to cross-ref content, and an automated way to list the content pages which have a reference to a specific tag. Say for example we have a list of notebooks, each of them has a "tag cell", in which the tags are listed. In markdown this would be done in the header:

Notebook 1:
```
---
tags: beginner mass-balance calibration
--- 
```

Notebook 2:
```
---
tags: advanced calibration
--- 
```

We'd want them to be displayed both in the actual notebook and in the rendered book.

Then the traditional book toc would be:

```
parts:
  - caption:  bla
    chapters:
    - file: notebook1
    - file: notebook2
```

But users could also search notebooks by "tag", i.e. the tag "calibration" would list both notebooks but the tag "advanced" would list only one.

Apologies if this has been suggested before!

**Comment by choldgraf** [comment]
Definitely something that has been discussed in other contexts but never fully implemented.

Cc @agoose77 who has discussed this in the context of mystmd.org . We are hoping build out an MVP of blogging or "collections" of pages somehow there.

One hack with the sphinx build system is to look into the ABlog extension, which brings blogging and tagging functionality to sphinx (and thus to jupyter book 1.X).

**Reply by fmaussion** [reply]
I'm just trying to revive this discussion a bit 😄 - I still think it would be super useful! Should I open an issue on the myst repo to keep track of this?
